container/container_test.go

```diff
@@ -0,0 +1,128 @@
+package container_test
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/cosmos/cosmos-sdk/container"
+)
+
+type KVStoreKey struct {
+	name string
+}
+
+type ModuleKey string
+
+type MsgClientA struct {
+	key ModuleKey
+}
+
+type KeeperA struct {
+	key KVStoreKey
+}
+
+type KeeperB struct {
+	key        KVStoreKey
+	msgClientA MsgClientA
+}
+
+type Handler struct {
+	Handle func()
+}
+
+type Command struct {
+	Run func()
+}
+
+func ProvideKVStoreKey(scope container.Scope) KVStoreKey {
+	return KVStoreKey{name: scope.Name()}
+}
+
+func ProvideModuleKey(scope container.Scope) ModuleKey {
+	return ModuleKey(scope.Name())
+}
+
+func ProvideMsgClientA(_ container.Scope, key ModuleKey) MsgClientA {
+	return MsgClientA{key}
+}
+
+type ModuleA struct{}
+
+func (ModuleA) Provide(key KVStoreKey) (KeeperA, Handler, Command) {
+	return KeeperA{key}, Handler{}, Command{}
+}
+
+type ModuleB struct{}
+
+type BDependencies struct {
+	container.StructArgs
+
+	Key KVStoreKey
+	A   MsgClientA
+}
+
+type BProvides struct {
+	KeeperB  KeeperB
+	Handler  Handler
+	Commands []Command
+}
+
+func (ModuleB) Provide(dependencies BDependencies) BProvides {
```

Comment:
is there a reason for using a struct here as returned value vs multiple returned values as in `ModuleA.Provide`?

Reply:
Just to give an example of using StructArgs